modules/ROOT/pages/external-oauth-2.0-token-validation-policy.adoc

```diff
@@ -1,43 +1,173 @@
+X-AGW-token_type=Bearer
+---
+
+The propagated properties do not include the `scope` and `expires_in` objects and arrays.
+
+For further processing within the flow, the original unparsed response from the validation endpoint is made available. The information is stored in an authentication object. 
+
+For example, if a `mail` field is returned to the policy, you can access the value of this field by using the following DataWeave 2.0 expression:
+
+`#[authentication.properties.userProperties.mail]`
+
+You can obtain the client ID of the OAuth2 token by using the following DataWeave 2.0 expression:
+
+`#[authentication.principal]`
+
+== Communicating with the Token Validation Endpoint Using a Proxy
+
+You can enable the OAuth 2.0 Access Token Enforcement using Mule OAuth Provider policy to use the gateway proxy settings by specify the following property when starting the Mule runtime engine: 
+
+`anypoint.platform.external_authentication_provider_enable_proxy_settings=<true|false(default)>`
+
+When you enable this property, the policy uses the Mule proxy settings if you have configured the following parameters:
+
+* `anypoint.platform.proxy_host=localhost`
+* `anypoint.platform.proxy_port=8080`
+
+=== Configuring Parameters for Mule Applications
+
+When you apply the policy to your API from the UI, a list of parameters is displayed:
+
+[%header%autowidth.spread,cols="30,40,30"]
+|===
+| Element | Description | Example
+| Scopes | Displays a space-separated list of supported scopes. | `READ`, `WRITE`, `READ and WRITE`
+| Scope Validation Criteria | Determines if the token must contain all defined scopes or just one.
+Values: `Contains all scopes` and `Contains any scope` | 
+If the ‘Contains all scopes’ value is selected, the tokens with the scopes `READ` and `WRITE` are accepted. If the ‘Contains any scopes’ value is selected, the tokens with the scope `READ`, `WRITE`, and `READ AND WRITE` are accepted.
+| Access Token validation endpoint url | The url of the Access Token validation endpoint of the External OAuth2 Provider. |The link to a mule provider hosted by the client: http://client-hostetd-mule-provider-url.cloudhub.io/validate
+| Validate TLS Certificate | Enables third-party authentication server TLS validation. | If selected, the communication with the token validation endpoint uses the TLS protocol.
+| Expose Headers | In a proxy scenario, specifies whether headers must be exposed in the request to the backend. User properties returned by the federation server are prefixed with 'X-AGW-' and sent as headers to the backend. | See the <<leveraging-TVE-information,Leveraging Token Validation Endpoint Information for Mule Applications>> for more information about this element.
+| Skip Client Id Validation | Skips the client application's API contract validation. | If selected, the Step 4 of the diagram in the <<how-this-policy-works,How This Policy Works>> section is skipped.
+|===
+
+== FAQs
+
+*Can I generate OAuth2 tokens with the policy?*
```

Comment:
GLOBALLY search for the characters "oauth" and verify that each represents the actual term presentation (OAuth, OAuth 2.0) or exactly what UI syntax has, in a syntax context and formatted as `syntax`.

Reply:
Is this correct:
```suggestion
*Can I generate OAuth 2.0 tokens with the policy?*
```

Reply:
Done! Thanks!